### PR TITLE
TGW FAQ and Spoke Alternate VPC Support

### PR DESCRIPTION
**Pull request by milldr**
## what
- Added FAQ to the TGW upgrade guide for replacing attachments
- Added note about destroying TGW components
- Added option to not create TGW propagation and association when connecting an alternate VPC

## why
- When connecting an alternate VPC in the same region as the primary VPC, we do not want to create a duplicate TGW propagation and association

## references
- n/a

### CODE REVIEW

File: modules/tgw/README.md [hunk 1]
```diff
@@ -340,3 +340,29 @@ tgw/spoke:
           stage: prod
           environment: usw2
 ```
+
+## Destruction
+
+When destroying Transit Gateway components, order of operations matters. Always destroy any removed `tgw/spoke` components before removing a connection from the `tgw/hub` component.
+
+The `tgw/hub` component creates map of VPC resources that each `tgw/spoke` component references. If the required reference is removed before the `tgw/spoke` is destroyed, Terraform will fail to destroy the given `tgw/spoke` component.
```
Comment (osterman) [inline modules/tgw/README.md:348]: Maybe consider suggesting using workflows to document how to create/destroy in the TGW network.

Comment (milldr) [reply modules/tgw/README.md:348]: added!

File: modules/tgw/README.md [hunk 2]
```diff
@@ -340,3 +340,57 @@ tgw/spoke:
...
+      - command: terraform destroy tgw/spoke -s plat-use1-dev --auto-approve
+      - command: terraform destroy tgw/spoke -s plat-use1-staging --auto-approve
+      - command: terraform destroy tgw/spoke -s plat-use1-prod --auto-approve
+      - command: echo 'Destroying core spokes for Transit Gateway'
+        type: shell
+        name: core-spokes
+      - command: terraform destroy tgw/spoke -s core-use1-auto --auto-approve
+      - command: terraform destroy tgw/spoke -s core-use1-network --auto-approve
+      - command: terraform destroy tgw/hub -s core-use1-network --auto-approve
+        name: hub
```
Comment (milldr) [inline modules/tgw/README.md:373]:
```suggestion
      - command: echo 'Destroying Transit Gateway Hub'
         type: shell
         name: hub
      - command: terraform destroy tgw/hub -s core-use1-network --auto-approve
```

File: modules/tgw/README.md [hunk 3]
```diff
@@ -340,3 +340,59 @@ tgw/spoke:
...
+      - command: terraform destroy tgw/spoke -s plat-use1-prod --auto-approve
+      - command: echo 'Destroying core spokes for Transit Gateway'
+        type: shell
+        name: core-spokes
+      - command: terraform destroy tgw/spoke -s core-use1-auto --auto-approve
+      - command: terraform destroy tgw/spoke -s core-use1-network --auto-approve
+      - command: echo 'Destroying Transit Gateway Hub'
+         type: shell
+         name: hub
+      - command: terraform destroy tgw/hub -s core-use1-network --auto-approve
```
Comment (milldr) [inline modules/tgw/README.md:375]:
```suggestion
      - command: echo 'Destroying Transit Gateway Hub'
        type: shell
        name: hub
      - command: terraform destroy tgw/hub -s core-use1-network --auto-approve
```

**Pull request merged**